### ISSUE TITLE
Drawer menu icon is not clickable when Android TalkBack screen reader is active

### ISSUE THREAD

**Issue by dmathisen** [react-navigation/react-navigation#12173]
### Current behavior

When Android "TalkBack" mode is enabled (screen reader accessibility feature), the Drawer menu icon is not clickable.

### Expected behavior

Users should be able to navigate the app with the screen reader enabled.

### Reproduction

https://snack.expo.dev/@winkleapp/drawer-navigation-|-react-navigation

See "Minimal example of drawer-based navigation" [here](https://reactnavigation.org/docs/drawer-based-navigation/), and run the [snack link](https://snack.expo.io/?platform=android&name=Drawer%20navigation%20%7C%20React%20Navigation&dependencies=%40expo%2Fvector-icons%40*%2C%40react-native-community%2Fmasked-view%40*%2Creact-native-gesture-handler%40*%2Creact-native-pager-view%40*%2Creact-native-paper%40%5E4.7.2%2Creact-native-reanimated%40*%2Creact-native-safe-area-context%40*%2Creact-native-screens%40*%2Creact-native-tab-view%40%5E3.0.0%2C%40react-navigation%2Fbottom-tabs%406.3.1%2C%40react-navigation%2Fdrawer%406.4.1%2C%40react-navigation%2Felements%401.3.3%2C%40react-navigation%2Fmaterial-bottom-tabs%406.2.1%2C%40react-navigation%2Fmaterial-top-tabs%406.2.1%2C%40react-navigation%2Fnative-stack%406.6.1%2C%40react-navigation%2Fnative%406.0.10%2C%40react-navigation%2Fstack%406.2.1&hideQueryParams=true&sourceUrl=https%3A%2F%2Freactnavigation.org%2Fexamples%2F6.x%2Fdrawer-based-navigation.js) on Android.

To reproduce:
1. Open the snack on Android (I'm on Pixel 6a, Android v14)
1. In Settings -> Accessibility -> Screen Reader -> Enable TalkBack (screen reader). Note: tap once to select UI elements and hear the description, then double tap to "click" the element. Use 2 fingers to swipe anything.
1. Everything on the screen is clickable EXCEPT for the hamburger menu icon. Nothing happens, making the app impossible to navigate.

Note: Ideally, the the screen reader would say something like "Menu icon to open drawer naviagation" when the user clicks the menu (using the accessibilityLabel prop).

### Platform

- [X] Android
- [ ] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [X] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [ ] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

- [x] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               | 6.1.18
| @react-navigation/drawer               | 6.7.2
| @react-navigation/native-stack         | 6.11.0
| react-native-safe-area-context         | 4.10.5
| react-native-screens                   | 3.31.1
| react-native-gesture-handler           | 2.16.1
| react-native-reanimated                | 3.10.1
| react-native                           | 0.74.5
| expo                                   | 51.0.34
| node                                   | 20.12.0
| npm                            | 10.8.3

**Comment by gianlucalippolis** [comment]
hi @dmathisen, i fixed it with this pull request. https://github.com/react-navigation/react-navigation/pull/12184

**Comment by dmathisen** [comment]
Amazing. Thank you!
Closing. I'll keep my eye on the next release and will re-open if I come across any issues.
Thanks again.

**Comment by dmathisen** [comment]
This should not be closed until the PR is merged, correct?
https://github.com/react-navigation/react-navigation/pull/12185/

CC @gianlucalippolis @satya164

**Comment by dmathisen** [comment]
Closing.
I cannot test because Expo doesn't support the version with the fix, but I can refer back to this if needed:
https://github.com/react-navigation/react-navigation/pull/12185#issuecomment-2483322149
